### ISSUE TITLE
Making BloomFilter polymophic

### ISSUE THREAD

**Issue by johandahlberg** [twitter/algebird#606]
Hi!

I've been playing around with making the Algebirds Bloom Filters polymorphic, and I'm wondering if there is any interest in a PR for this?

As it stands right now I've implemented this using a implicit view from `A => Array[Byte]`, making things look something like this (translating from the usage example on the website):

```
implicit def stringToBytes(s: String): Array[Byte] = s.getBytes()

val NUM_HASHES = 6
val WIDTH = 32

val bfMonoid = new BloomFilterMonoid[String](NUM_HASHES, WIDTH)
val bf = bfMonoid.create("1", "2", "3", "4", "100")

val approxBool = bf.contains("1")
val res = approxBool.isTrue
```

I think it's kind of neat, however I completely understand if this is not something folks here are interested in since it would introduce a breaking change to the API. This could of course be worked around by making `BloomFilterMonoid` still be bound to `String`, but be backed up by the polymorphic implementation. Before looking into that, I'd be curious to know if there is any interest in making the Bloom Filters in making the Bloom Filters polymorphic in the first place

Cheers,
Johan

**Comment by johnynek** [comment]
Yeah, bloom filter needs a lot of work. We should do what you describe as was done for CMS and HLL.

We should also optimize bloom filter more. It was a contribution that has not received the attention that other aspects of the code have.

We should add benchmarks and improve them.

Lastly, it would be interesting to see if we could remove the dependency on java ewah. I'm not sure it is actually faster than a smallish local and immutable implementation.

**Comment by sritchie** [comment]
@johandahlberg We'd love a PR with these ideas if you're interested!

**Comment by johandahlberg** [comment]
This was closed by #607.
